### PR TITLE
feat: Add useAppLinkWithStoreFallback hook

### PR DESCRIPTION
**Pull request by Crash--**
Add the hook already used by Harvest and now Drive. It's time to add it to cozy-client. 

The goal of this hook is to give the link to the app or to the store if it's not installed yet 

I know we have the request status issue see https://github.com/cozy/cozy-client/issues/646

### CODE REVIEW

File: packages/cozy-client/src/node.js [hunk 1]
```diff
@@ -31,3 +31,5 @@ export * from './cli'
 
 import * as models from './models'
 export { models }
+
+export * from './hooks'
```
Comment (y-lohse) [inline packages/cozy-client/src/node.js]: Why do we want to export hooks for the node version?

Comment (Crash--) [reply packages/cozy-client/src/node.js]: Currently, jest is based on our `main` entry point aka our node entry and not our browser one.

Comment (y-lohse) [reply packages/cozy-client/src/node.js]: Mmh ok, but that means `react` is imported on node again, and I think that was a problem?

Comment (Crash--) [reply packages/cozy-client/src/node.js]: fixed https://github.com/cozy/cozy-client/pull/651/commits/1b049d71996fe7ef7f065bf433c7ac649a824ef7

File: package.json [hunk 2]
```diff
@@ -34,19 +34,43 @@
...
-    ],
-    "setupFilesAfterEnv": [
-      "<rootDir>/packages/cozy-stack-client/src/__tests__/setup.js"
-    ],
-    "testMatch": [
-      "**/(*.)(spec|test).js?(x)"
-    ],
-    "modulePathIgnorePatterns": [
-      "<rootDir>/packages/.*/dist/"
+    "projects": [
```
Comment (y-lohse) [inline package.json]: TIL

File: package.json [hunk 3]
```diff
@@ -34,19 +34,43 @@
...
+          "jest-localstorage-mock"
+        ],
+        "setupFilesAfterEnv": [
+          "<rootDir>/packages/cozy-stack-client/src/__tests__/setup.js"
+        ],
+        "browser": true,
+        "modulePathIgnorePatterns": [
+          "<rootDir>/packages/.*/dist/"
+        ]
+      }
```
Comment (ptbrowne) [inline package.json]: Most of the configs are shared, could they be regrouped in a commonConfig and then a `merge` could be used to only change what differs between the two ?

Comment (Crash--) [reply package.json]: https://github.com/cozy/cozy-client/pull/651/commits/6cab6ed73a8fec3b53a17e898ddc8d00c6c9ebdc done thanks :)

**Pull request merged**